### ISSUE TITLE
Marionette does not wait that page will have been loaded

### ISSUE THREAD

**Issue by drobota** [mozilla/geckodriver#96]
I noticed that marionette does not wait that page will be loaded. It starts immediately to look for elements on the page.

Webdriver: geckodriver-0.8.0-OSX
Firefox: 47, 48
Selenium: 2.53.5

**Comment by AkosLukacs** [comment]
I can confirm this, and also it doesn't wait for page reloads either.

Maybe this is a different issue, but if you 
 - load a page, that contains input fields, 
 - do a refresh (I actually set a cookie before the refresh), 
 - find an input box, and start "typing" into the input box
Firefox actually finds and fills out the input box before the refresh is complete.

The same code works fine in IE and Chrome.

OS: Win10 64bit
Webdriver: geckodriver-0.9.0, win64 (same behaviour with 0.8)
Firefox (developer edition): 49 and lower
Selenium: 2.53.1 (C# driver)

**Comment by NotThatGupta** [comment]
Are there any updates on this? 

I have ran into this issue as well and came across this thread: http://stackoverflow.com/questions/38857031/selenium-firefox-marionette-driver-not-doing-wait-until.

Seems to be the same issue that I am having. Running the same code using an older selenium version on an older version of Firefox works perfectly, but if I want it to work with marionette, I have to specify explicit waits many times when clicking on anything that triggers a modal, expands a menu, or opens a new page.

Could someone please confirm whether this is a known issue with implicit waits or whether there is some other change that needs to be made to get scrips to run with gecko driver?

Thanks!

**Comment by andreastt** [comment]
I believe possibly this bug is addressed (in part) by [bug 1307127](https://bugzilla.mozilla.org/show_bug.cgi?id=1307127).

@NotThatGupta The WebDriver specification does not say we should do implicit waits on clicking or anything that implies an action in the page. There is an [ongoing discussion](https://lists.w3.org/Archives/Public/public-browser-tools-testing/2016OctDec/thread.html) in the WG right now whether we should.

**Comment by NotThatGupta** [comment]
Thanks @andreastt.

I read through the discussion and I don't know where you guys ended up, but I hope that you guys implement implicit waits for at least click, send keys, and get text. I think this would be a significant breaking change for a lot of folks. 

Just my 2 cents...can I follow that discussion to stay up to date on what you guys decide in the end? Thanks again!

**Comment by NotThatGupta** [comment]
@andreastt Any updates on the discussion regarding implicit wait? Last post I see says that voting was extended through last week, but I don't see any results.

**Comment by hjwp** [comment]
Dupe of #308 ?

**Comment by andreastt** [comment]
It is indeed a duplicate of https://github.com/mozilla/geckodriver/issues/308.

**Comment by drobota** [comment]
Indeed #308 is dup. When this is original issue which had been reported 4 month before #308
